### PR TITLE
update: Update rules.

### PR DESCRIPTION
**Pull request by jakubmazanec**

### CODE REVIEW

**Review by ecl1ps — APPROVED** [review]

File: packages/config-eslint/src/rules/import.ts [hunk 1]
```diff
@@ -12,32 +12,34 @@ const config: ESLintConfig['rules'] = {
...
-  ], // enforce a convention in module import order
-  'import/prefer-default-export': 'off', // prefer a default export if module exports a single name
   'import/no-extraneous-dependencies': [
     'error',
     {
       devDependencies: [
         `test/**/*.${JSX_EXTS_GROUP}`,
         `tests/**/*.${JSX_EXTS_GROUP}`,
         `**/*.test.${JSX_EXTS_GROUP}`,
+        `**/__tests__/**/*.${JSX_EXTS_GROUP}`,
```
Comment (rajzik) [inline packages/config-eslint/src/rules/import.ts:22]: Please add `__tests__/**/*.${JSX_EXTS_GROUP}` as well.

File: packages/config-eslint/src/rules/typescript.ts [hunk 2]
```diff
@@ -52,14 +53,14 @@ const config: ESLintConfig['rules'] = {
...
   ], // ensure consistent use of file extension within the import path
-
   'import/no-extraneous-dependencies': [
     'error',
     {
       devDependencies: [
         `test/**/*.${TJSX_EXTS_GROUP}`,
         `tests/**/*.${TJSX_EXTS_GROUP}`,
         `**/*.test.${TJSX_EXTS_GROUP}`,
+        `**/__tests__/**/*.${TJSX_EXTS_GROUP}`,
```
Comment (rajzik) [inline packages/config-eslint/src/rules/typescript.ts:63]: same as above.

File: packages/config-eslint/src/rules/typescript.ts [hunk 3]
```diff
@@ -142,7 +143,7 @@ const config: ESLintConfig['rules'] = {
       leadingUnderscore: 'forbid',
       trailingUnderscore: 'forbid',
       filter: {
-        regex: '^_+$',
+        regex: '^(_+|__webpack_public_path__)$',
         match: false,
       },
```
Comment (rajzik) [inline packages/config-eslint/src/rules/typescript.ts:148]: is it possible to add `_anything` to the function definition? Or how does it work right now? 
```tsx
(_context, theme) => theme.color;
```

File: packages/config-eslint/src/rules/typescript.ts [hunk 4]
```diff
@@ -298,7 +299,7 @@ const config: ESLintConfig['rules'] = {
   '@typescript-eslint/no-throw-literal': 'error', // disallow throwing literals as exceptions
   '@typescript-eslint/no-type-alias': 'off', // disallow the use of type aliases
   '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error', // flags unnecessary equality comparisons against boolean literals
-  '@typescript-eslint/no-unnecessary-condition': ['error', { allowConstantLoopConditions: true }], // prevents conditionals where the type is always truthy or always falsy -- false positive for some cases
```
Comment (rajzik) [inline packages/config-eslint/src/rules/typescript.ts:301]: Can you move these to some block with `todos` and `fixmes`?

**Review by rajzik — CHANGES_REQUESTED** [review]

File: packages/config-eslint/src/rules/import.ts (hunk 1 above)
Comment (ecl1ps) [reply packages/config-eslint/src/rules/import.ts:22]: I think that `**/__tests__/**/*.${JSX_EXTS_GROUP}` is superset of `__tests__/**/*.${JSX_EXTS_GROUP}`

File: packages/config-eslint/src/rules/typescript.ts (hunk 4 above)
Comment (jakubmazanec) [reply packages/config-eslint/src/rules/typescript.ts:301]: I don't know what you mean.

File: packages/config-eslint/src/rules/typescript.ts (hunk 3 above)
Comment (jakubmazanec) [reply packages/config-eslint/src/rules/typescript.ts:148]: `(_, __, ___) => { /* ... */ }` etc. is allowed (the rule config works this way: if identifier matches a @typescript-eslint/naming-convention selector, in this case "variableLike", it is then checked against its regexp and if it matches the regexp as well, the identifier is ignored by the linter rule).
Anyway, just use `context` in your example (I think TypeScript config for unused parameters should be off, as it is useless).

File: packages/config-eslint/src/rules/import.ts (hunk 1 above)
Comment (jakubmazanec) [reply packages/config-eslint/src/rules/import.ts:23]: Anyway, done.

File: packages/config-eslint/src/rules/typescript.ts (hunk 2 above)
Comment (jakubmazanec) [reply packages/config-eslint/src/rules/typescript.ts:64]: Done.

File: packages/config-eslint/package.json [hunk 5]
```diff
@@ -20,6 +20,9 @@
   "engines": {
     "node": ">=10.16.0"
   },
+  "resolutions": {
```
[the remark this answers is not in the document]
Comment (jakubmazanec) [reply packages/config-eslint/package.json:23]: Import plugin is broken and it doesn't that it will be fixed soon.

**Review by rajzik — APPROVED** [review]

**Pull request merged**